tools/validators/instance_validator/validate/entity_instance.py

```diff
@@ -15,293 +15,391 @@
 """Uses ontology universe to validate parsed instance data."""
```

Comment:
there is duplicate code in this file similar to the one in build/lib/validate.

What do you think about the following:

EntityInstance(raw_yaml)
--> identifies the raw 'translation' section and delegates to translation.py to extract the yaml and validate which also populates the fields (in OOP manner)
-----> translation.py identifies the raw 'point' section and delegates to point.py to extract the yaml and validation and populate the fields (in OOP manner)
-----> translation.py identifies the raw 'states' section and delegates to states.py to extract the yaml and validation and populate the fields (in OOP manner)
-----> translation.py identifies the raw 'units' section and delegates to units.py to extract the yaml and validation and populate the fields (in OOP manner)
--> identifies the raw 'connection' section and delegates to connection.py to extract the yaml and validate which also populates the fields (in OOP manner)


More info in BuildingConfiguration.java (on cs/DigitalBuildings)

Reply:
I think you have a point about inconsistency in the location of parsing code vs. object model code vs. validation code.

I see two ways to resolve it. There's the way you're suggesting in this comment, where each file defining a class is responsible for parsing and validation for that class. And then there's the possibility of creating entity_parser.py and entity_validator.py which encapsulate the parsing and validation logic respectively.

The first option (the one you suggest in this comment) seems a bit better to me, because it makes it easier to extend the object model if each class is fully responsible for the logic relating to that type of object.

However, this PR is already pretty large. Do you mind if I add a TODO and leave this as a next step in the refactoring effort?

Reply:
Sounds good, thx